.travis.yml

```diff
@@ -0,0 +1,56 @@
+# Use new trusty images, should yield newer compilers and packages
+sudo: true
+dist: trusty
+language: cpp
+
+matrix:
+  include:
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-4.9
+      env:
+         - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-7
+      env:
+        - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
+
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - llvm-toolchain-trusty-3.9
+          packages:
+            - clang-3.9
+      env:
+        - MATRIX_EVAL="CC=clang-3.9 && CXX=clang++-3.9"
+
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - llvm-toolchain-trusty-5.0
+          packages:
+            - clang-5.0
+      env:
+        - MATRIX_EVAL="CC=clang-5.0 && CXX=clang++-5.0"
+
+before_install:
+  - sudo apt-get update -qq
+  # imake
+  - sudo apt-get install -y xutils-dev
```

Comment:
Why do we need this? These are build-time dependencies, so they should be installed automatically by the environment (i.e., `pbuilder` or `sbuild` or whatever Travis uses), right?

Besides, I'm not exactly convinced that our build environment handles `CC` and `CXX` env flags correctly (mostly because this was never tested so far IIRC). That's not a problem with your PR, but something that should be kept in mind and investigated in a build log to see if they actually take correctly.

Reply:
I wanted to keep it simple, but we could make this as complex as you want to. 

About the `CC` and `CXX`, I agree with you that this could be not working 100% OK, and should be IMHO in another PR. The same rules with the many warnings that the diferent compilers are reporting.